ERCS/erc-7527.md

```diff
@@ -0,0 +1,510 @@
+---
+eip: 7527
+title: Token Bound Function Oracle AMM Contract
+description: An interface system that wrap/unwrap between fungible tokens and non-fungible tokens based on an embedded Function Oracle AMM
+author: Elaine Zhang (@lanyinzly) <lz8aj@virginia.edu>, Jerry <jerrymindflow@gmail.com>, Amandafanny <amandafanny200@gmail.com>, Shouhao Wong (@wangshouh) <wongshouhao@outlook.com>, 0xPoet <0xpoets@gmail.com>
+discussions-to: https://ethereum-magicians.org/t/eip-7527-token-bound-function-oracle-amm-contract/15950
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-09-03
+requires: 165, 721
+---
+
+
+## Abstract
+
+This proposal outlines a system that incorporates a Function Oracle capable of converting fungible tokens into a single non-fungible token each instance and then reverting the non-fungible token back to fungible tokens. The pre-configured Function Oracle comes equipped with its own Automated Market Maker (AMM), facilitating a clearing mechanism for voucher issuers.
```

Comment:
Generally, goal for an ERC is not for creating a "system" but for interfaces and behaviors that's going to be shared amongst different implementations. 
```suggestion
This proposal outlines interfaces for a Function Oracle capable of converting fungible tokens into a single non-fungible token each instance and then reverting the non-fungible token back to fungible tokens. The pre-configured Function Oracle comes equipped with its own Automated Market Maker (AMM) interfaces, enabling a clearing mechanism for voucher issuers.
```